tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py

```diff
@@ -0,0 +1,289 @@
+        client1.exec_command(
+            sudo=True,
+            cmd="sed -i 's/step take root1/step take root2/' crush.map_root2.txt",
+        )
+        client1.exec_command(
+            sudo=True, cmd="crushtool -c crush.map_root2.txt -o crush_root2.map.bin"
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph osd setcrushmap -i crush_root2.map.bin"
+        )
+
+        time.sleep(120)
+        df_utilization_root1_step_2, rc = client1.exec_command(
+            sudo=True, cmd="ceph osd df root1 -f json"
+        )
+        df_utilization_root1_step_2 = json.loads(df_utilization_root1_step_2)
+
+        df_utilization_root2_step_2, rc = client1.exec_command(
+            sudo=True, cmd="ceph osd df root2 -f json"
+        )
+        df_utilization_root2_step_2 = json.loads(df_utilization_root2_step_2)
+        log.info("Step 2 df values for root1 and root2")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root1")
+        log.info(f"\n{out}")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root2")
+        log.info(f"\n{out}")
+        if (
+            df_utilization_root1_step_1["summary"]["total_kb_used"]
+            < df_utilization_root1_step_2["summary"]["total_kb_used"]
+        ):
+            raise CommandFailed(
+                f"Data not migrated to the root2 root1 before :"
+                f"{df_utilization_root1_step_1['summary']['total_kb_used']}"
+                f" is greater than after:{df_utilization_root1_step_2['summary']['total_kb_used']}"
+            )
+        log.info(
+            f"data has been migrated from root1 \n before: "
+            f"{df_utilization_root1_step_1['summary']['total_kb_used']}"
+            f"\n After :{df_utilization_root1_step_2['summary']['total_kb_used']}"
+        )
+
+        if (
+            df_utilization_root2_step_2["summary"]["total_kb_used"]
+            < df_utilization_root2_step_1["summary"]["total_kb_used"]
+        ):
+            raise CommandFailed(
+                f"Data migrated to different root than root2 \nbefore : "
+                f"{df_utilization_root2_step_1['summary']['total_kb_used']}, "
+                f"After it is :{df_utilization_root2_step_2['summary']['total_kb_used']}"
+            )
+        log.info(
+            f"data has been migrated to root2 \n before: "
+            f"{df_utilization_root2_step_1['summary']['total_kb_used']}"
+            f"\n After :{df_utilization_root2_step_2['summary']['total_kb_used']}"
+        )
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        fs_util.client_clean_up(
+            "umount", fuse_clients=[clients[0]], mounting_dir=fuse_mounting_dir_1
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph osd setcrushmap -i crush.map_initial.bin"
+        )
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd tree")
```

Comment:
Can the additional OSD pools created be released to ensure no impact on next test? Restoring the original config in OSD and CephFS context?

Reply:
We are not creating any pools. we are using cephfs data pool only.
All OSDs changes will be reverted back when we apply **crush.map_initial.bin** file